### PR TITLE
Refactor some Merkle tree related methods

### PR DESCRIPTION
**Pull request by Uxio0**
- Refactor some merkle_proof methods
- Add merkle proof test

### CODE REVIEW

File: vestings/merkle_proof.py [hunk 1]
```diff
@@ -19,38 +21,33 @@ def combine_and_hash(leaf1: str, leaf2: str):
...
-            if int.from_bytes(
-                bytes(HexBytes(leaf1)), byteorder="big", signed=False
-            ) < int.from_bytes(bytes(HexBytes(leaf2)), byteorder="big", signed=False):
-                current_level.append(combine_and_hash(leaf1, leaf2))
-            else:
-                current_level.append(combine_and_hash(leaf2, leaf1))
-            i = i + 2
+            leaf2 = previous_level[i + 1] if i + 1 < count else EMPTY_HASH
+            if HexBytes(leaf1) > HexBytes(leaf2):
+                leaf1, leaf2 = leaf2, leaf1
```
Comment (rmeissner) [inline vestings/merkle_proof.py:35]: Maybe a short comment here would be nice. It is an important line that is easy to miss.

Comment (Uxio0) [reply vestings/merkle_proof.py:35]: Yeah, I'm still refactoring the method as I don't like the while loop

Comment (Uxio0) [reply vestings/merkle_proof.py:31]: Updated

File: vestings/merkle_proof.py [hunk 2]
```diff
@@ -1,103 +1,64 @@
...
+
+    # ValueError is raised if element is not found
+    element_index = vestings_tree[0].index(element)
 
-    level_count = len(vestings_tree)
-    while current_level_index < level_count - 1:
-        current_level = vestings_tree[current_level_index]
+    proof: List[HexStr] = []
+    for current_level in vestings_tree[:-1]:
+        # If element in pair position, get next leave, or EMPTY_HASH if there's not
```
Comment (elgatovital) [inline vestings/merkle_proof.py:48]: have you meant `leaf` here?

Comment (Uxio0) [reply vestings/merkle_proof.py]: Sure, thanks, fixed!

**Review by elgatovital — APPROVED** [review]

**Pull request merged**